sentry-tracing/src/converters.rs

```diff
@@ -22,93 +24,99 @@ pub fn convert_tracing_level(level: &tracing_core::Level) -> Level {
 }
 
 /// Extracts the message and metadata from an event
-pub fn extract_event_data(
-    event: &tracing_core::Event,
-) -> (Option<String>, BTreeMap<String, Value>) {
+fn extract_event_data(event: &tracing_core::Event) -> (Option<String>, FieldVisitor) {
     // Find message of the event, if any
-    let mut data = BTreeMapRecorder::default();
-    event.record(&mut data);
-    let message = data
-        .0
+    let mut visitor = FieldVisitor::default();
+    event.record(&mut visitor);
+    let message = visitor
+        .json_values
         .remove("message")
         .map(|v| v.as_str().map(|s| s.to_owned()))
         .flatten();
-
-    (message, data.0)
+    (message, visitor)
 }
 
 /// Extracts the message and metadata from a span
-pub fn extract_span_data(attrs: &span::Attributes) -> (Option<String>, BTreeMap<String, Value>) {
-    let mut data = BTreeMapRecorder::default();
+pub(crate) fn extract_span_data(
+    attrs: &span::Attributes,
+) -> (Option<String>, BTreeMap<String, Value>) {
+    let mut data = FieldVisitor::default();
     attrs.record(&mut data);
 
     // Find message of the span, if any
     let message = data
-        .0
+        .json_values
         .remove("message")
         .map(|v| v.as_str().map(|s| s.to_owned()))
         .flatten();
 
-    (message, data.0)
+    (message, data.json_values)
 }
 
-#[derive(Default)]
 /// Records all fields of [`tracing_core::Event`] for easy access
-struct BTreeMapRecorder(pub BTreeMap<String, Value>);
+#[derive(Default)]
+struct FieldVisitor {
+    pub json_values: BTreeMap<String, Value>,
+    pub exceptions: Vec<Exception>,
+}
 
-impl BTreeMapRecorder {
+impl FieldVisitor {
     fn record<T: Into<Value>>(&mut self, field: &Field, value: T) {
-        self.0.insert(field.name().to_owned(), value.into());
+        self.json_values
+            .insert(field.name().to_owned(), value.into());
     }
 }
 
-impl Visit for BTreeMapRecorder {
-    fn record_debug(&mut self, field: &Field, value: &dyn std::fmt::Debug) {
-        self.record(field, format!("{:?}", value));
-    }
+impl Visit for FieldVisitor {
     fn record_i64(&mut self, field: &Field, value: i64) {
         self.record(field, value);
     }
+
     fn record_u64(&mut self, field: &Field, value: u64) {
         self.record(field, value);
     }
+
     fn record_bool(&mut self, field: &Field, value: bool) {
         self.record(field, value);
     }
+
     fn record_str(&mut self, field: &Field, value: &str) {
         self.record(field, value);
     }
+
+    fn record_error(&mut self, _field: &Field, value: &(dyn Error + 'static)) {
+        let event = event_from_error(value);
+        for exception in event.exception {
```

Comment:
We should probably provide a better API for this, but this will do for now.

Reply:
You could create a separate issue for it, if you have a description in mind. "Sentry event exceptions should not be exposed through a public field?"